python/xgbserver/xgbserver/model.py

```diff
@@ -22,29 +22,41 @@
 
 
 class XGBoostModel(kfserving.KFModel):
-    def __init__(self, name: str, model_dir: str, nthread: int,
+    def __init__(self, name: str, model_dir: str, nthread: int, method: str,
                  booster: XGBModel = None):
         super().__init__(name)
         self.name = name
         self.model_dir = model_dir
         self.nthread = nthread
+        self.method = method
+        self._classifier = None
         if not booster is None:
             self._booster = booster
             self.ready = True
 
     def load(self) -> bool:
         model_file = os.path.join(
             kfserving.Storage.download(self.model_dir), BOOSTER_FILE)
-        self._booster = xgb.Booster(params={"nthread": self.nthread},
-                                    model_file=model_file)
+        if self.method == "predict_proba":
+            self._classifier = xgb.XGBClassifier()
+            self._classifier.load_model(model_file)
```

Comment:
This is currently kind of a hack. Since we can't deduct from XGB Boosters which model it was (can be either `XGBClassifier()` or `XGBRTClassifier()` for `predict_proba`) we load it in to an `XGBClassifier`. This means we don't support `predict_proba` for the other model type.